mondrian/src/main/java/mondrian/spi/Dialect.java

```diff
@@ -161,6 +161,18 @@ String quoteIdentifier(
         String name);
 
     /**
+     * Encloses an identifier in quotation marks appropriate for the
+     * current SQL dialect. For example, in Oracle, where the identifiers
+     * are quoted using double-quotes,
+     * <code>quoteIdentifier("schema.name")</code> yields a string
+     * containing <code>"schema"."name"</code>.
+
+     * @param name Name to be quoted.
+     * @return Quoted identifier
+     */
+    String quoteIdentifierWithInnerDots(String name);
```

Comment:
I don't like adding a new method just for this one call. The logic that decides what to do with dots should be transparent to the caller and the dialect itself should hide that logic. I recommend removing this method and handling everything in Dialect.quoteIdentifier